RecoPixelVertexing/PixelVertexFinding/src/gpuClusterTracks.h

```diff
@@ -0,0 +1,243 @@
+#ifndef RecoPixelVertexing_PixelVertexFinding_clusterTracks_H
+#define RecoPixelVertexing_PixelVertexFinding_clusterTracks_H
+
+#include<cstdint>
+#include<cmath>
+#include<cassert>
+
+#include "HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h"
+
+#include "gpuVertexFinder.h"
+
+namespace gpuVertexFinder {
+  
+  // this algo does not really scale as it works in a single block...
+  // enough for <10K tracks we have
+  __global__ 
+  void clusterTracks(int nt,
+		     OnGPU * pdata,
+		     int minT,  // min number of neighbours to be "core"
+		     float eps, // max absolute distance to cluster
+		     float errmax, // max error to be "seed"
+		     float chi2max   // max normalized distance to cluster
+		     )  {
+
+    constexpr bool verbose = false;
+    
+    auto er2mx = errmax*errmax;
+    
+    auto & data = *pdata;
+    float const * zt = data.zt;
+    float const * ezt2 = data.ezt2;
+    float * zv = data.zv;
+    float * wv = data.wv;
+    float * chi2 = data.chi2;
+    uint32_t & nv = *data.nv;
+    
+    int8_t  * izt = data.izt;
+    int32_t * nn = data.nn;
+    int32_t * iv = data.iv;
+    
+    assert(pdata);
+    assert(zt);
+    
+    __shared__ HistoContainer<int8_t,8,5,8,uint16_t> hist;
+    
+    //  if(0==threadIdx.x) printf("params %d %f\n",minT,eps);    
+    //  if(0==threadIdx.x) printf("booked hist with %d bins, size %d for %d tracks\n",hist.nbins(),hist.binSize(),nt);
+    
+    // zero hist
+    hist.nspills = 0;
```

Comment:
can this be wrapped in some member function ?
and, shouldn't it be done by a single thread (or spreading the bins through the threads) ?